spinta/utils/response.py

```diff
@@ -106,7 +108,58 @@
 
         if request.method == 'POST':
             context.bind('transaction', manifest.backend.transaction, write=True)
-            data = await request.json()
+
+            # only `application/json` is supported
+            ct = request.headers.get('content-type')
+            if ct != 'application/json':
+                return JSONResponse(
+                    {"error": "only 'application/json' content-type is supported"},
+                    status_code=415
+                )
+
+            auth_header = request.headers.get('authorization')
+            if auth_header is None:
+                return JSONResponse({"error": "missing authorization header"},
+                                    status_code=401)
+
+            if not auth_header.startswith('Bearer '):
+                return JSONResponse({"error": "invalid authorization header"},
+                                    status_code=403)
+
+            jwt_token = auth_header.strip('Bearer ') if auth_header else None
+
+            # TODO: check for special scope which allows id creation
+            # scopes = jwt.decode(jwt_token)
+            #
+            # XXX: currently for dev purposes and testing use
+            # fake token `f00b4rb4z`
+            can_create_ids = bool(jwt_token == "f00b4rb4z")
+
+            # make sure json is valid
+            try:
+                data = await request.json()
+            except json.decoder.JSONDecodeError:
+                return JSONResponse({"error": "not a valid json"},
+                                    status_code=400)
+
+            object_is_being_created = ('id' in data.keys() and not params.id)
+            if object_is_being_created:
+                if not can_create_ids:
+                    return JSONResponse({"error": "cannot create 'id'"},
+                                        status_code=400)
+                else:
+                    try:
+                        id = commands.get(context, model, model.backend, data['id'])
+                    except (MultipleRowsException, NoResultsException):
+                        pass
+                    else:
+                        return JSONResponse({"error": "cannot create duplicate 'id'"},
+                                            status_code=400)
```

Comment:
Wrong error message, and you don't need to check that here, this is already checked by `check` command in `spinta.backends.postgresql.check`.